### PR TITLE
Use MainCoroutineRule in tests

### PR DESCRIPTION
**Pull request by NotWoods**

### CODE REVIEW

File: app/src/migration/java/org/mozilla/fenix/MigratingFenixApplication.kt [hunk 1]
```diff
@@ -7,6 +7,7 @@ package org.mozilla.fenix
 import android.content.Context
 import kotlinx.coroutines.runBlocking
 import mozilla.components.support.migration.FennecMigrator
+import org.mozilla.fenix.migration.MigrationTelemetryListener
```
Comment (eliserichards) [inline app/src/migration/java/org/mozilla/fenix/MigratingFenixApplication.kt:10]: Why does this file need the telemetry listener?

File: app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt [hunk 2]
```diff
@@ -38,7 +36,9 @@ import mozilla.components.feature.tab.collections.Tab as ComponentTab
 @ExperimentalCoroutinesApi
 class DefaultSessionControlControllerTest {
 
-    private val mainThreadSurrogate = newSingleThreadContext("UI thread")
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule(TestCoroutineDispatcher())
+
```
Comment (eliserichards) [inline app/src/test/java/org/mozilla/fenix/home/DefaultSessionControlControllerTest.kt:41]: 👏

File: app/src/migration/java/org/mozilla/fenix/MigratingFenixApplication.kt (hunk 1 above)
Comment (NotWoods) [reply app/src/migration/java/org/mozilla/fenix/MigratingFenixApplication.kt:10]: It [already uses it](https://github.com/NotWoods/fenix/blob/use-main-coroutine-rule/app/src/migration/java/org/mozilla/fenix/MigratingFenixApplication.kt#L53), I'm just fixing the package name here

**Comment by NotWoods** [comment]
Closing because this was fixed in the Kotlin dependency bump

**Pull request closed without merging**